### DISCUSSION TITLE
Code untuk mengisi data-test kolom yang kosong dengan hasil forecast (FnB)

### DISCUSSION THREAD

**Discussion by AjiPramasto** [teamalgoritma/community#331] [Q&A] [answered]
Halo,

terakait dengan capstone FnB,
sudah dilakukan predict untuk mengisi kolom di data-test, namun hasilnya seperti ini:
<img width="634" alt="image" src="https://user-images.githubusercontent.com/107389614/191708196-ee8a84cc-7fd0-41da-8528-ce95355a03c6.png">
kode yang dipakai: pred <- predict(ts_train_forecast, test)
pertanyaannya:
kolom waktu ko berbeda ya formatnya?mohon bantuannya

sedangkan untuk ARIMA, sudah dibuat tsdisplay hasilnya seperti ini:
<img width="634" alt="image" src="https://user-images.githubusercontent.com/107389614/191709046-ea020f3f-0b9f-4b4c-862a-02a0cc1a8bd5.png">
nah menentukan PACF dan ACFnya bagaimana ya?

**Comment by NabiilahArdini** [comment, marked as answer]
Dear @AjiPramasto, hasil prediksi menampilkan `Point Forecast` berupa prediksi titik forecast (ini adalah target kita), dan batas atas batas bawah untuk confidence interval 80% dan 90%. Untuk langsung mengambil `Point Forecast` nya, silahkan gunakan `pred$mean`.

Untuk model ARIMA, pada fungsi `tsdisplay()`, pastikan data yang digunakan adalah data setelah di differencing sejumlah yang dibutuhkan, sampai data stasioner. Selain itu, sepertinya lag yang ditampilkan sangat banyak (sampai ratusan). Silahkan gunakan parameter `lag.max` pada fungsi `ts_display()`:
* `lag.max` = jumlah lag maximum yang ingin dilihat, contoh sepanjang 3x seasonal atau dikisaran 15 (apabila data tanpa seasonal) 

Dari sana, Semoga Bapak dapat menentukan ACF dan PACF dengan lebih mudah. 

Demikian yang dapat saya sarankan. Semoga informasi ini dapat bermanfaat.  
Apabila comment ini menjawab pertanyaan Bapak, silahkan klik "Mark as Answer" pada comment ini.

**Comment by AjiPramasto** [comment]
Dear Ka @NabiilahArdini ,

terima kasih input dan sarannya, namun ada beberapa hal yang masih belum mengerti/paham:
1. kode untuk menggunakan pred$mean bagaimana ya Ka?ini untuk filling nilai di data-test kan ya?
    sebelumnya kode untuk membuat TS sbb:
<img width="620" alt="image" src="https://user-images.githubusercontent.com/107389614/192180530-26004c1f-600c-4c53-b0b4-38b413a4b40d.png">
sehingga timenya menjadi seperti ini:
<img width="524" alt="image" src="https://user-images.githubusercontent.com/107389614/192180589-8186bed9-121b-4578-aac0-8f2618f7c741.png">

nah itu berpengaruh tidak ya untuk kolom datetime di data-test?

2. kode SARIMA belum berhasil --> menggunakan SARIMA karena di data terdapat seasonal dan trend:
<img width="643" alt="image" src="https://user-images.githubusercontent.com/107389614/192180748-0e918e1e-b8a4-4127-aa46-75e04b6a5161.png">
 
mohon bantuannya dan penjelasannya.

3. untuk mengukur accuracy forecast data-test bagaimana ya Ka?karena kan data actual data-test tidak ada.

mohon bantuannya.
terima kasih sebelumnya.

**Reply by NabiilahArdini** [reply]
1. Untuk hasil decomposenya sudah tepat ya. Tidak ada pengaruh buruk terhadap data_test. Kemudian `pred$mean` didapatkan dari hasil `forecast()` yang Bapak lakukan sebelumnya, itu adalah hasil forecasting menggunakan model yang Bapak pilih. Nilai tersebut dapat digunakan untuk mengisi test.csv yang disediakan oleh Algoritma:
```
# prediction
pred <- forecast(model, h = <sepanjang tests.csv>)
pred$mean

# input to data test
data_test<- data_test %>% mutate(visitor = pred$mean)
write.csv(data.test, "<file path dan nama file submission>")
````

2. Saya belum menemukan penjelasan juga. Namun apakah mungkin karena Bapak menggunakan multiseasonal data ke ARIMA yang biasa? bagaimana jika Bapak menggunakan fungsi `stlm()` saja untuk pemodelan ARIMA nya. Berikut contohnya: https://rpubs.com/nabiilahardini/ts-flight

```
# ETS/Holt-Winters
air_ets <- stlm(air_train, method = "ets", lambda = 0) # lambda = 0 for no log transformation for addivite data
# SARIMA
air_arima <- stlm(air_train, method = "arima", lambda = 0)
```

3. Untuk mengetahui error data test perlu disubmit hasil prediksi nya ke [Algoritma Capstone Leaderboard](https://algoritma-caps.shinyapps.io/leaderboard/). Silahkan mengikuti panduan di [Rpubs Capstone ML](https://rpubs.com/AlgoritmaAcademy/ml-capstone) pada bagian Submission.

Demikian yang dapat kami infokan. Semoga bermanfaat!
